### PR TITLE
[FIX] website_sale: fix price for product carousel

### PR DESCRIPTION
**Pull request by myhy-odoo**
Problem:
The prices for products shown on the dynamic content: products on the website can be displayed incorrectly.
Example:
Consider Product AB that is shared between Company A and Company B. Company A sets Tax A for Product AB.
Company B sets Tax B for Product AB.
Product AB is published on Website A that belongs to Company A. However, the price shown for Product AB on the dynamic content: products is incorrect because the price includes taxes from both Company A & B.

Solution:
Filter the tax records based on the current context's allowed_company_ids. The price for the product should only include taxes based on the website's company.

opw-3201283
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale/models/product_product.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def _get_contextual_price_tax_selection(self):
         self.ensure_one()
         price = self._get_contextual_price()
         line_tax_type = self.env['ir.config_parameter'].sudo().get_param('account.show_line_subtotals_tax_selection')
-        if line_tax_type == "tax_included" and self.taxes_id:
-            price = self.taxes_id.compute_all(price, product=self, partner=self.env['res.partner'])['total_included']
+        company_taxes = self.taxes_id.filtered(lambda tax: tax.company_id.id in self._context.get('allowed_company_ids'))
```
Comment (Feyensv) [inline addons/website_sale/models/product_product.py:103]:
```suggestion
        company_taxes = self.taxes_id.filtered(lambda tax: tax.company_id == self.env.company)
```

the context key `allowed_company_ids` should not be managed in business code unless really necessary.  Use the environment company instead (which encapsulates the `allowed_company_ids` logic and provides a fallback when the context key is not specified).

**Review by Feyensv — COMMENTED** [review]
Little remark, makes sense to discard taxes from other companies here :+1:.

File: addons/website_sale/models/product_product.py (hunk 1 above)
Comment (myhy-odoo) [reply addons/website_sale/models/product_product.py:103]: I made the change to the commit. Thank you for the info!

**Review by Feyensv — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**